src/main/webapp/app/lecture/lecture-unit/lecture-unit-management/video-unit-form/video-unit-form.component.ts

```diff
@@ -13,31 +13,44 @@ export interface VideoUnitFormData {
     competencies?: Competency[];
 }
 
-function videoUrlValidator(control: AbstractControl) {
-    if (control.value === undefined || control.value === null || control.value === '') {
-        return null;
-    }
-
-    const videoInfo = urlParser.parse(control.value);
-    return videoInfo ? null : { invalidVideoUrl: true };
+function isTumLiveUrl(url: URL): boolean {
+    return url.host === 'live.rbg.tum.de';
 }
 
-function urlValidator(control: AbstractControl) {
-    let validUrl = true;
+function isVideoOnlyTumUrl(url: URL): boolean {
+    return url?.searchParams.get('video_only') === '1';
+}
 
-    // for certain cases like embed links for vimeo
-    const regex = /^\/\/.*$/;
-    if (control.value && control.value.match(regex)) {
-        return null;
+function videoSourceTransformUrlValidator(control: AbstractControl): ValidationErrors | undefined {
+    const urlValue = control.value;
+    if (!urlValue) {
+        return undefined;
     }
-
+    let parsedUrl, url;
     try {
-        new URL(control.value);
+        url = new URL(urlValue);
+        parsedUrl = urlParser.parse(urlValue);
```

Comment:
(Nitpick) again, this **could** be put after the try-catch block as it just returns undefined if the parsedUrl is not valid.